hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/replication/health/MismatchedReplicasHandler.java

```diff
@@ -57,35 +58,28 @@ public MismatchedReplicasHandler(
    */
   @Override
   public boolean handle(ContainerCheckRequest request) {
-    ContainerInfo containerInfo = request.getContainerInfo();
-    Set<ContainerReplica> replicas = request.getContainerReplicas();
+    if (request.isReadOnly()) {
+      return false;
+    }
+
+    final ContainerInfo containerInfo = request.getContainerInfo();
+    final Set<ContainerReplica> replicas = request.getContainerReplicas();
+
     if (containerInfo.getState() != HddsProtos.LifeCycleState.CLOSED &&
         containerInfo.getState() != HddsProtos.LifeCycleState.QUASI_CLOSED) {
       // Handler is only relevant for CLOSED or QUASI-CLOSED containers.
       return false;
     }
-    LOG.debug("Checking container {} in MismatchedReplicasHandler",
-        containerInfo);
 
-    if (request.isReadOnly()) {
-      return false;
-    }
-    // close replica if needed
-    for (ContainerReplica replica : replicas) {
-      if (shouldBeClosed(containerInfo, replica)) {
-        LOG.debug("Sending close command for mismatched replica {} of " +
-            "container {}.", replica, containerInfo);
+    LOG.debug("Checking container {} in MismatchedReplicasHandler", containerInfo);
 
-        if (containerInfo.getState() == HddsProtos.LifeCycleState.CLOSED) {
-          replicationManager.sendCloseContainerReplicaCommand(
-              containerInfo, replica.getDatanodeDetails(), true);
-        } else if (containerInfo.getState() ==
-            HddsProtos.LifeCycleState.QUASI_CLOSED) {
-          replicationManager.sendCloseContainerReplicaCommand(
-              containerInfo, replica.getDatanodeDetails(), false);
-        }
-      }
-    }
+    final Predicate<ContainerReplica> shouldSendClose = (r) -> shouldSendClose(containerInfo, r);
+
+    replicas.stream().filter(shouldSendClose).forEach(r -> {
+      LOG.debug("Sending close command for mismatched replica {} of container {}.", r, containerInfo);
```

Comment:
We can have info log for sending close in this handle, including force, replica state and container state. This will be useful for debugging.

Reply:
`replicationManager.sendCloseContainerReplicaCommand` ends up calling `ReplicationManager#sendDatanodeCommand` which already logs the command details (including force flag) and the ContainerInfo at `info` level.

Adding another info level log here will be redundant.

Here at debug level, the additional information that's getting logged is `replica` info.